### DISCUSSION TITLE
docs/core/gdb-config-file

### DISCUSSION THREAD

**Discussion by giscus** [gogf/gf-site#38] [General]
# docs/core/gdb-config-file

我们推荐使用配置组件来管理数据库配置，并使用 g 对象管理模块中的 g.DB("数据库分组名称") 方法获取数据库操作对象，数据库对象将会自动读取配置组件中的相应配置项，并自动初始化该数据库操作的单例对象。数据库配置管理功能使用的是配置管理组件实现（配置组件采用接口化设计默认使用文件系统实现），同样支持多种数据格式如： toml/yaml/json/xml/ini/properties。默认并且推荐的配置文件数据格式为 yaml。

https://goframe.org/docs/core/gdb-config-file

**Comment by LRLHHC** [comment]
请问有没有类似 gorm  .Debug() 方法可以单独打印某一句 sql 呢？

**Comment by black1552** [comment]
https://www.bookstack.cn/read/GoFrame-2.0-zh/4fec5162cfb4d90e.md#%3Ccode%3EUnscoped%3C/code%3E%E5%BF%BD%E7%95%A5%E6%97%B6%E9%97%B4%E7%89%B9%E6%80%A7   自动插入创建时间与更新时间的时候  创建时间 插入的时间比正常时间少了一天  而 更新时间在每次更新的时候比上次插入时间减少了几个小时 是怎么回事

**Comment by lz-freedom** [comment]
orm 部分的日志能不能自定义驱动，实现结构化打印，例如下面这段日志，打印成
{
    "du": 62,
    "duf": "62 ms",
    "rows": 1,
    "sql": "UPDATE `sys_admin_user` SET `last_active_at`=null,`update_time`='2025-04-21 10: 20: 02' WHERE (`id`=1) AND `delete_time` IS NULL",
}
[ 62 ms] [default] [badazhou_saas] [rows:1  ] UPDATE `sys_admin_user` SET `last_active_at`=null,`update_time`='2025-04-21 10:20:02' WHERE (`id`=1) AND `delete_time` IS NULL

**Comment by ccyuki** [comment]
请问，咱们这里是不是不支持 DB 的故障转移？
我理解这里的主从只是用于负载均衡和读写分离，假设如果有 3 个 DB（其中一个异常），那么当 dao.table.... 操作时，如果连接报错，不会自动基于其他两个 DB 进行重试。

**Comment by beidanghan** [comment]
什么时候支持doris数据库，  因为借用mysql协议的时候，有时使用orm操作函数会产生不可预料的错误，一般我都是使用原生sql使用，特别麻烦。咱这边什么时候准备支持，有计划吗？

**Comment by zhyfor** [comment]
希望增加sql慢日志的配置及打印
